### PR TITLE
GH-153: Command responses don't get posted in the RHS 

### PR DESCRIPTION
**Pull request by tw-ayush**
added the rootId in server post request from args.

#153

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ dist
 .vscode
 
 assets/i18n/translate.*.json
+/.idea/
```
Comment (hanzei) [inline .gitignore:10]: Would you please remove this unrelated change?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution :+1: 
A few questions:
1. Does the post in `settingsError` also need to get assigned to a thread?
2. And what about the post in https://github.com/mattermost/mattermost-plugin-jitsi/pull/158/files#diff-7278f3371df5ef589578d375bde66febR307-R316?

File: .gitignore (hunk 1 above)
Comment (tw-ayush) [reply .gitignore]: sure done.

**Comment by tw-ayush** [comment]
> 1. settingsError

Amm i guess you are right, I'll add the errors as well in the RHS. 

I did not understand the second point.

**Comment by hanzei** [comment]
At the end of `executeSettingsCommand` there is another Post that gets build. It's roughly Line 300. Does this also need an update?

**Comment by tw-ayush** [comment]
> At the end of `executeSettingsCommand` there is another Post that gets build. It's roughly Line 300. Does this also need an update?

changes done 👍

**Comment by hanzei** [comment]
Thanks for updating the PR :+1: There are also two other instances of `model.Post` in `plugin.go`. Could you please also take a look at them? Sorry, I missed them in the first place.

**Comment by tw-ayush** [comment]
> Thanks for updating the PR 👍 There are also two other instances of `model.Post` in `plugin.go`...

No worries I also did not do a complete sweep through.
Now the problem with doing these changes in `plugin.go` is that those methods are called from two places once inside `command.go` another `api.go` and the latter does not have `model.CommandArgs` in it.

**Comment by hanzei** [comment]
You can change the signature to pass a rootID as it's called in https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/command.go#L140

**Comment by tw-ayush** [comment]
> You can change the signature to pass a rootID as it's called in
>
can i also add the rootId in 
https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/api.go#L30

, would it break the integration or just pass in "" (empty string) from `api.go`.

**Comment by hanzei** [comment]
The code path in `api.go` is triggered when the meeting is created from the UI which is always in the center channel. Hence passing an empty string is fine.

**Comment by tw-ayush** [comment]
I believe this should do fine. Please validate the PR now.

**Review by hanzei — APPROVED** [review]
Great work :+1: 


While testing I found one uncovered edge case: If the naming schema is set to `ask`, the ephemeral message generated when typing `/jitsi start` is correct posted in the RHS, but the following meeting isn't.

Given that this an edge case, I'm fine with merging this PR as it is and opening a new issue for that.

But if you also want to fix that, let me know!

**Comment by tw-ayush** [comment]
hey can [this](https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/plugin.go#L204) be the issue ? 

and yes sure assign it to me i'll have a look.

**Comment by hanzei** [comment]
I don't think so. `handleStartMeeting` need go get the rootId from `StartMeetingFromAction`. Either `StartMeetingFromAction` in https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/api.go#L167 already contains that information or you have to add it to that struct.

**Review by larkox — APPROVED** [review]
Apart of @hanzei 's comments, LGTM. Thanks!

**Comment by tw-ayush** [comment]
> mattermost-plugin-jitsi/server/api.go

looks like changing [here](https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/api.go#L232) would fix it 
https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/api.go#L232

**Comment by hanzei** [comment]
I think you need to touch https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/api.go#L224. https://github.com/mattermost/mattermost-plugin-jitsi/blob/14107b6a514180ca0867761c0cba72bcc77518ac/server/api.go#L232 is for the case that the webapp button was used.

**Comment by hanzei** [comment]
@tw-ayush Would you mind merging this PR as it is and I open a new ticket for the outstanding issue?

**Comment by tw-ayush** [comment]
> @tw-ayush Would you mind merging this PR as it is and I open a new ticket for the outstanding issue?

I'm a bit confused here.. aren't you going to merge this?

**Comment by hanzei** [comment]
@DHaussermann I've tested this PR in depth my self. Let me know if you also want to test it.

**Comment by hanzei** [comment]
@tw-ayush Surely I want to merge this PR. Let me queue it for QA testing.

I opened https://github.com/mattermost/mattermost-plugin-jitsi/issues/160 for the edge case I talked about in https://github.com/mattermost/mattermost-plugin-jitsi/pull/158#pullrequestreview-506103114.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Ephemeral posts made by Jitsi bot are shown in RHS as expected
- Tested various post including help and config changes
- Brief regression testing

LGTM!

Thanks @tw-ayush for this enhancement!
